### PR TITLE
[aws] Remove hardcoded event.dataset field and use ECS field instead

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## Proposed commit message

1. This PR is to remove hardcoded `event.dataset` field and reference to the ECS field instead. 
The value for `event.dataset` is already applied from the elastic-agent side, please see this [PR](https://github.com/elastic/beats/pull/20076) for more detail.

With this change, when user tries to use a customized `dataset` value, `event.dataset` should change as well.

~~2. This PR also changes the default dataset name under Advanced option from `generic` to `aws.cloudwatch_logs`.~~
This is removed from the PR because I misunderstood why default is `generic`. In theory logs coming from cloudwatch should be rerouted to different data streams based on which service the log comes from. Whatever left in the default datastream are the ones not being routed. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes https://github.com/elastic/integrations/issues/8810

### CODE REVIEW

**Review by andrewkroh — COMMENTED** [review]
Q: If the only goal is to remove the hard-coded value, could the field data type still be retained as constant_keyword for performance reasons? The field value in the first event determines its value within a backing index. `data_stream.dataset` works this way.

You can still use the ECS definition with a type override:

```
- external: ecs
  name: event.dataset
  type: constant_keyword
```

File: packages/aws/data_stream/cloudwatch_logs/manifest.yml [hunk 1]
```diff
@@ -123,7 +123,7 @@ streams:
       - name: data_stream.dataset
         type: text
         required: true
-        default: generic
+        default: aws.cloudwatch_logs
```
[the remark this answers is not in the document]
Comment (kaiyan-sheng) [reply packages/aws/data_stream/cloudwatch_logs/manifest.yml:126]: Good point! I need to revert this part and reconsider! Thanks!

**Comment by agithomas** [comment]
@tommyers-elastic , 

Should we not make similar changes in all Integration packages?

**Comment by kaiyan-sheng** [comment]
@andrewkroh The issue that led to this bug requires `event.dataset` to have more than one value so we can't use `constant_keyword` here. The original log message already has an `event.dataset` field and we are adding one more on top.

Do you know in ECS is that why we are keeping `event.dataset` to be `keyword` instead of `constant_keyword` or multiple values for the same `event.dataset` actually make sense?

**Comment by kaiyan-sheng** [comment]
+1 @agithomas That's also my next question.

**Comment by andrewkroh** [comment]
In #8810 it says

> This field should be added by the [agent](https://github.com/elastic/beats/pull/20076) so we should remove hardcoded value for `event.dataset` in all integrations.

And it implies that `event.dataset` is being set the same way as `data_stream.dataset` (via processors added by Elastic Agent). So then why is it OK for `data_stream.dataset` to be a constant_keyword (without a static value in the mapping), but not for `event.dataset`? Is there some other follow on processing that I'm unaware of that changes `event.dataset` but leaves `data_stream.dataset` untouched?

If a data stream handles multiplexed data like perhaps cloudwatch logs does and the event.dataset is customized based on the log type then this would make sense to relax the constant_keyword type for that data stream. But for something like cloudtrail that only handles cloudtrail data I don't see any reason to relax the constant_keyword type for event.dataset.

**Comment by kaiyan-sheng** [comment]
> If a data stream handles multiplexed data like perhaps cloudwatch logs does and the event.dataset is customized based on the log type then this would make sense to relax the constant_keyword type for that data stream. But for something like cloudtrail that only handles cloudtrail data I don't see any reason to relax the constant_keyword type for event.dataset.

Yes! Sorry I was only thinking about cloudwatch logs in the previous comment. Yes I will keep `event.dataset` to be `keyword` type in `cloudwatch_logs` data stream. The rest of the data streams I'm adding `type: constant_keyword`.

**Review by aspacca — APPROVED** [review]

**Review by andrewkroh — APPROVED** [review]

**Comment by gizas** [comment]
@kaiyan-sheng following list can be consider a candidate not to have type: constant_keyword?
cloudfront_logs
route53_public_logs
route53_resoolver_logs
firewall_logs

Also for my undertasting the ` type: keyword` is not needed in `cloudwatch_logs`,  is it assumed?

**Review by gizas — APPROVED** [review]

**Comment by kaiyan-sheng** [comment]
> @kaiyan-sheng following list can be consider a candidate not to have type: constant_keyword? cloudfront_logs route53_public_logs route53_resoolver_logs firewall_logs
> 
> Also for my undertasting the ` type: keyword` is not needed in `cloudwatch_logs`, is it assumed?

@gizas Yes we don't need to add `type: keyword` for `cloudwatch_logs` because it's already the type defined in ECS for `event.dataset`. And we add `type: constant_keyword` to the rest of the log type data streams because cloudwatch log data stream is the only one user can use to ingest different kinds of logs, like an input.

**Review by agithomas — APPROVED** [review]
LGTM!

**Pull request merged**